language/go/fileinfo.go

```diff
@@ -569,8 +569,24 @@ func checkConstraints(c *config.Config, os, arch, osSuffix, archSuffix string, t
 	}
 
 	goConf := getGoConfig(c)
-	checker := func(tag string, ts tagSet) bool {
-		if isIgnoredTag(tag) {
+
+	isIgnoredTag := func(tag string) bool {
+		return goConf.genericTags[tag]
+	}
+
+	if tags != nil && tags.expr != nil {
+		replacedExpr, err := dropNegationForIgnoredTags(tags.expr, isIgnoredTag)
+		if err != nil {
+			panic(fmt.Sprintf("unexpected error while dropping negation for generic tags: %s", err))
```

Comment:
Let's do one of two things here:

1. `log.Printf` the error here instead of panicking.
2. Push the `panic` down into `dropNegationForIgnoredTags` and change the signature of that function and `newBuildTags` to not return an error.

My preference is (2) but I don't want to expand scope on you too much. (1) is less work.

This condition only matters if the syntax for build constraints changes. That seems unlikely to me at this point, and panicking is a reasonable way to handle that.

Reply:
I've pushed a change implementing (2), which of course does make the handling of these conditions more consistent.